cmd/notification.go

```diff
@@ -1753,19 +1754,28 @@ func (sys *NotificationSys) DriveSpeedTest(ctx context.Context, opts madmin.Driv
 				resp.Error = err.Error()
 			}
 
-			ch <- resp
+			select {
+			case <-ctx.Done():
+			case ch <- resp:
+			}
 
 			reqInfo := (&logger.ReqInfo{}).AppendTags("remotePeer", client.host.String())
 			ctx := logger.SetReqInfo(GlobalContext, reqInfo)
 			logger.LogIf(ctx, err)
 		}(client)
 	}
 
+	wg.Add(1)
 	go func() {
-		wg.Wait()
-		close(ch)
+		defer wg.Done()
+		select {
+		case <-ctx.Done():
+		case ch <- driveSpeedTest(ctx, opts):
+		}
 	}()
 
+	wg.Wait()
```

Comment:
DriveSpeedTest() is an async function, it is supposed to return 'ch' and the caller consumes it. We should not wait here

Reply:
yes my mistake, making the change @vadmeste